ddl/ddl_api.go

```diff
@@ -1333,10 +1335,6 @@ func (d *ddl) AddColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTab
 
 // AddTablePartitions will add a new partition to the table.
 func (d *ddl) AddTablePartitions(ctx sessionctx.Context, ident ast.Ident, spec *ast.AlterTableSpec) error {
-	if len(spec.PartDefinitions) == 0 {
```

Comment:
Why you remove this？What if the Type == model.PartitionTypeRange and the len(spec.PartDefinitions) is 0?